runtime/src/account_rent_state.rs

```diff
@@ -162,21 +190,98 @@ mod tests {
+        assert!(post_rent_state.transition_allowed_from(
+            &RentState::Uninitialized,
+            restrict_rent_paying_accounts_ttl_extension
+        ));
+        assert!(post_rent_state.transition_allowed_from(
+            &RentState::RentExempt,
+            restrict_rent_paying_accounts_ttl_extension
+        ));
+        assert!(post_rent_state.transition_allowed_from(
+            &RentState::RentPaying {
+                data_size: 0,
+                lamports: 1,
+            },
+            restrict_rent_paying_accounts_ttl_extension
+        ));
 
         let post_rent_state = RentState::RentExempt;
-        assert!(post_rent_state.transition_allowed_from(&RentState::Uninitialized));
-        assert!(post_rent_state.transition_allowed_from(&RentState::RentExempt));
-        assert!(post_rent_state.transition_allowed_from(&RentState::RentPaying(0)));
-
-        let post_rent_state = RentState::RentPaying(2);
-        assert!(!post_rent_state.transition_allowed_from(&RentState::Uninitialized));
-        assert!(!post_rent_state.transition_allowed_from(&RentState::RentExempt));
-        assert!(!post_rent_state.transition_allowed_from(&RentState::RentPaying(3)));
-        assert!(!post_rent_state.transition_allowed_from(&RentState::RentPaying(1)));
-        assert!(post_rent_state.transition_allowed_from(&RentState::RentPaying(2)));
+        assert!(post_rent_state.transition_allowed_from(
+            &RentState::Uninitialized,
+            restrict_rent_paying_accounts_ttl_extension
+        ));
+        assert!(post_rent_state.transition_allowed_from(
+            &RentState::RentExempt,
+            restrict_rent_paying_accounts_ttl_extension
+        ));
+        assert!(post_rent_state.transition_allowed_from(
+            &RentState::RentPaying {
+                data_size: 0,
+                lamports: 1,
+            },
+            restrict_rent_paying_accounts_ttl_extension
+        ));
+        let post_rent_state = RentState::RentPaying {
+            data_size: 2,
+            lamports: 5,
+        };
+        assert!(!post_rent_state.transition_allowed_from(
+            &RentState::Uninitialized,
+            restrict_rent_paying_accounts_ttl_extension
+        ));
+        assert!(!post_rent_state.transition_allowed_from(
+            &RentState::RentExempt,
+            restrict_rent_paying_accounts_ttl_extension
+        ));
+        assert!(!post_rent_state.transition_allowed_from(
+            &RentState::RentPaying {
+                data_size: 3,
+                lamports: 5
+            },
+            restrict_rent_paying_accounts_ttl_extension
+        ));
+        assert!(!post_rent_state.transition_allowed_from(
+            &RentState::RentPaying {
+                data_size: 1,
+                lamports: 5
+            },
+            restrict_rent_paying_accounts_ttl_extension
+        ));
+        // No resize, same lamports is always allowed.
```

Comment:
I understand that the word "transition" is implied here, but if you could find a way to rewrite these comments so they don't say "lamports is" so many times, I would appreciate it.